ext/rbkit_event_packer.c

```diff
@@ -207,6 +211,52 @@ static void pack_object_space_dump_event(rbkit_object_space_dump_event *event, m
   }
 }
 
+static void pack_cpu_sample_event(rbkit_cpu_sample_event *event, msgpack_packer *packer) {
+  rbkit_cpu_sample *sample;
+
+  msgpack_pack_map(packer, 3);
+  sample = event->sample;
+
+  // Keys 1 & 2 - event type and timestamp
+  pack_event_header(packer, event->event_header.event_type);
+
+  // Key 3 : Payload
+  msgpack_pack_int(packer, rbkit_message_field_payload);
+  // Value 3: Array of samples
+  msgpack_pack_array(packer, sample->frame_count);
+
+  if(sample->frame_count != 0) {
```

Comment:
no need of if statement. for loop written properly should take care of this.
`for (size_t count = 0; count != sample->frame_count; ++count)`